### PR TITLE
set `Current height` as locktime for `Contract` & `Funding` tx.

### LINKED ISSUE #229: Use Current Height as locktime to prevent `Fee-Snipping Attacks`
* For `Fedility` & normal transactions  -> we are setting locktime as current height for preventing `fee-snipping attack` .
https://github.com/citadel-tech/coinswap/blob/691c937454ccf1c2a182ef0aca1d514b7f74606d/src/wallet/fidelity.rs#L354

* However, there is no locktime set for the `funding` and `contract` transactions, which is not ideal. These transactions are vital components of the protocol, and any attack on them could hinder the protocol's efficiency.

https://github.com/citadel-tech/coinswap/blob/691c937454ccf1c2a182ef0aca1d514b7f74606d/src/wallet/funding.rs#L177
https://github.com/citadel-tech/coinswap/blob/691c937454ccf1c2a182ef0aca1d514b7f74606d/src/protocol/contract.rs#L397

* They also must have current-height as locktime to prevent this attack .

### PR DESCRIPTION
**Pull request by KnowWhoami**
Fix #229

### CODE REVIEW

**Review by mojoX911 — COMMENTED** [review]
Approach Ack. Something broke in the tx construction which is failing the test.

File: src/wallet/direct_send.rs [hunk 1]
```diff
@@ -215,7 +215,8 @@ impl Wallet {
         }
 
         // Set the Anti-Fee-Snipping locktime
-        let lock_time = LockTime::from_height(self.rpc.get_block_count().unwrap() as u32).unwrap();
+        let current_height = self.rpc.get_block_count().unwrap();
+        let lock_time = LockTime::from_height(current_height as u32).unwrap();
```
Comment (Shourya742) [inline src/wallet/direct_send.rs:219]: Remove these unwraps..

File: src/wallet/funding.rs [hunk 2]
```diff
@@ -171,10 +171,16 @@ impl Wallet {
                     script_sig: ScriptBuf::new(),
                 })
                 .collect::<Vec<_>>();
+
+            // Set the Anti-Fee-Snipping locktime
+            let current_height = self.rpc.get_block_count()?;
+
+            let lock_time = LockTime::from_height(current_height as u32).unwrap();
```
Comment (Shourya742) [inline src/wallet/funding.rs:178]: Same for this

**Review by Shourya742 — CHANGES_REQUESTED** [review]
Concept ACK. I will through review this soon. Need to get around the idea of snipping. Till then just remove the unwraps and propagate them.

File: src/wallet/swapcoin.rs [hunk 3]
```diff
@@ -844,7 +846,8 @@ mod tests {
         let tx = Transaction {
             input: vec![input.clone()],
             output: vec![],
-            lock_time: LockTime::ZERO,
+            // TODO: left to check.
```
Comment (mojoX911) [inline src/wallet/swapcoin.rs:849]: This TODO still applies?

**Review by mojoX911 — COMMENTED** [review]
ACK.. Looking good. minor nits as mentioned by @Shourya742

File: src/protocol/contract.rs [hunk 4]
```diff
@@ -521,6 +521,8 @@ mod test {
     use core::panic;
     use std::str::FromStr;
 
+    const TEST_CURRENT_HEIGHT: u32 = 100;
```
[the remark this answers is not in the document]
Comment (mojoX911) [reply src/protocol/contract.rs:524]: It doesn't matter. Yes generally const are architectural components. In tests you can just hard code the value. Nothing is affected in runtime either way.

File: src/wallet/swapcoin.rs [hunk 5]
```diff
@@ -671,6 +671,8 @@ mod tests {
     use super::*;
     use bitcoin::{NetworkKind, PrivateKey};
 
+    const TEST_CURRENT_HEIGHT: u32 = 100;
```
[the remark this answers is not in the document]
Comment (mojoX911) [reply src/wallet/swapcoin.rs:674]: For the sake of consistency, it makes sense. Doesn't impact anything serious if we don't.

**Review by mojoX911 — APPROVED** [review]
ACK 02333ec946d3ff0b153937d5e7e90f13f144ad35

**Comment by mojoX911** [comment]
@KnowWhoami Green from me to merge. 

If you have more improve ideas can be done in a follow-up PR. 

@Shourya742 if it looks good to you we can merge.

**Review by Shourya742 — APPROVED** [review]
ACK

**Pull request merged**